### DISCUSSION TITLE
[cargo] from "=0.x.0" to "=0.y.0" considered "minor" instead of "major"

### DISCUSSION THREAD

**Discussion by Oliboy50** [renovatebot/renovate#20069] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.113

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub Enterprise Server 3.7.4

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

👋 Thank you so much for maintaining Renovate. ❤️ 

### Problem

As you can see on the following screenshot, Renovate considers that moving Cargo dependencies from `=0.21.0` to `=0.23.0` and from `=0.13.1` to `=0.21.0` are `minor` changes, whereas it is common knowledge in the Rust world that moving from `0.x` to `0.y` may break the public API of the crate and therefore should be considered `major` changes.

![Capture d’écran 2023-01-27 à 08 31 41](https://user-images.githubusercontent.com/2571084/215039711-541f91b4-7d49-4134-892c-e658ceb0a391.png)

![Capture d’écran 2023-01-27 à 10 09 21](https://user-images.githubusercontent.com/2571084/215049565-6fd20b06-6c9a-44fb-83da-db42f2b03233.png)

### Relevant Renovate config
```
{
    "packageRules": [
        {
            "matchManagers": ["cargo"],
            "matchUpdateTypes": [
                "minor",
                "patch"
            ],
            "groupName": "all cargo non-major dependencies",
            "groupSlug": "cargo-all-minor-patch"
        }
    ]
}
```

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
Please see https://github.com/renovatebot/renovate/discussions/19923 for a recent discussion on the same topic in Composer

In short: breaking is not the same thing as major. A (potentially) breaking 0.x to 0.x+1 release is still not a major release.

Renovate is correct, and this is not a bug, so I have converted it to a discussion.

Your expectations are otherwise fine though, as it's good to treat potentially breaking releases with caution. You can achieve this with package rules.

**Reply by Oliboy50** [reply]
Thanks for the quick answer :)

I'll ask there if they have a working example of how I could achieve that myself using `packageRules` Renovate config.

**Reply by rarkins** [reply]
Try matchCurrentVersion="<1"

**Reply by Oliboy50** [reply]
If I still want to group `minor` and `patch` (include `0.x.y+1` but exclude `0.x+1` updates) in a single PR, is it valid to use the same `groupName` for both rules, such as:

```json
{
    "packageRules": [
        {
            "groupName": "cargo all non breaking changes",
            "matchManagers": ["cargo"],
            "matchUpdateTypes": ["minor", "patch"]
        },
        {
            "groupName": "cargo all non breaking changes",
            "matchManagers": ["cargo"],
            "matchUpdateTypes": ["minor"],
            "matchCurrentVersion": "<1"
        }
    ]
}
```

❓

**Reply by rarkins** [reply]
Yes it's valid to do that but I don't think the logic above is correct? I would have expected something like:

```
{
    "packageRules": [
        {
            "matchManagers": ["cargo"],
            "matchUpdateTypes": ["patch"],
            "groupName": "cargo all non breaking changes"
        },
        {
            "matchManagers": ["cargo"],
            "matchUpdateTypes": ["minor"],
            "matchCurrentVersion": ">=1.0.0",
            "groupName": "cargo all non breaking changes"
        }
    ]
}
```

BTW I prefer putting matching first and actions like grouping last because the flow is better when reading it, but it doesn't make a functional difference.

**Reply by Oliboy50** [reply]
indeed, thank you so much for your time ❤️
